Bundle add files to requested datapack.

jasebell opened the pull request:
- New view (bundle-add)

- New hander to send the update to datastore.

- Activity finished handler to send info message back to user.

Commit:
Updates from rebase.

acron0 commented on src/cljs/witan/ui/controllers/datastore.cljs:
```diff
@@ -694,6 +694,24 @@
+  [event {:keys [groups message]}]
+  #_(activities/start-activity!
+     :send-collect-request
+     (data/new-command!)))
```
I think this is a merge relic.

jasebell replied:
Removed.

acron0 commented on src/cljs/witan/ui/strings.cljs:
```diff
@@ -325,6 +327,8 @@
    :string.activity.send-collect-request/failed         "You failed to send a collection request for datapack '%s'"
    :string.activity.send-collect-request/completed      "You successfully sent a collection request to %s recipient(s), for datapack '%s'"
+   :string.activity.add-files-to-datapack/failed "You failed to add files to the datapack."
+   :string.activity.add-files-to-datapack/completed "You successfully added files to the datapack."
```
There are already failed and completed strings that look good (`string.activity.add-file-to-datapack/completed`) - can we use these?

jasebell replied:
I don't think we can use those as they are formatted strings, C+S bundle add has no detail on the datapack name at this point as it's not loaded in for the view.

acron0 replied:
You're correct - my bad

acron0 commented on src/cljs/witan/ui/controllers/datastore.cljs:
```diff
@@ -951,6 +969,16 @@
+  [:add-collect-files-to-datapack :failed]
+  [{:keys [args]}]
+  (.info js/toastr (:log args)))
+
```
Once the activity finishes, what happens? What does the user see or where do they go?

acron0 commented on src/cljs/witan/ui/components/bundle-add.cljs:
```diff
@@ -0,0 +1,106 @@
+                         :prevent? true
+                         :disabled? (empty? @files)}
+                        #(controller/raise! :data/add-collect-files-to-datapack {:added-files @files
+                                                                                 :datapack-id (:id (:route/params (data/get-app-state :app/route)))}))]]])))
```
The whole form needs disabling whilst the activity is in progress, to stop people hammering the button, changing files etc. Also, a message if/when it fails/succeeds.

Commit:
Bundle add page work.

- Added bundle-add schema in app state for pending/messages.
- Additional CSS for padded buttons on the page.
- Handler and Finished function calls for adding files to datapack.
- Handling empty ids
- Route additions

Commit:
Change incorrect failure message name.

Commit:
Remove errornous character in function name.
